diff --git a/components/esp32/Kconfig b/components/esp32/Kconfig
index 754ca607b8..7ca4b727c7 100644
--- a/components/esp32/Kconfig
+++ b/components/esp32/Kconfig
@@ -700,7 +700,6 @@ menu "ESP32-specific"
 
     choice ESP32_RTC_EXT_CRYST_ADDIT_CURRENT_METHOD
         prompt "Additional current for external 32kHz crystal"
-        depends on ESP32_RTC_CLK_SRC_EXT_CRYS
         depends on ESP32_REV_MIN_FULL < 200
         default ESP32_RTC_EXT_CRYST_ADDIT_CURRENT_NONE
         help
